src/collect-all/Utilities.cs

```diff
@@ -0,0 +1,479 @@
+         {
+            if (!Directory.Exists(directory))
+            {
+               throw new DirectoryNotFoundException($"{directory} not found.");
+            }
+         }         
+
+         if (!Directory.Exists(coreFxNativeBinDir))
+         {
+            throw new DirectoryNotFoundException($"{coreFxNativeBinDir} not found.");
+         }
+         
+         this.m_corefxBinDir = coreFxBinDir;
+         this.m_nativeCoreFxBinDir = coreFxNativeBinDir;
+      }
+      
+      // Set Core_Root for superPMI.
+     
+      if (this.m_loggingVerbose)
+      {
+         Console.WriteLine(" ");
+         Console.WriteLine($"CORE_ROOT: {this.m_coreRoot}");
+         Console.WriteLine(" ");
+      }
+     
+      Environment.SetEnvironmentVariable("CORE_ROOT", this.m_coreRoot);
+
+   }
+   
+   /////////////////////////////////////////////////////////////////////////////
+   // Public Methods
+   /////////////////////////////////////////////////////////////////////////////
+   
+   public static Utilities GetInstance(string tempPath = null, bool crossGen = false, bool loggingVerbose = false, string arch = null, string buildType = null, string coreclrRepoLocation = null)
+   {
+      if (Utilities.m_Instance == null)
+      {
+         Utilities.m_Instance = new Utilities(tempPath, crossGen, loggingVerbose, arch, buildType, coreclrRepoLocation);
+      }
+      
+      return Utilities.m_Instance;
+   }
+   
+   /////////////////////////////////////////////////////////////////////////////
+   // Utility Methods
+   /////////////////////////////////////////////////////////////////////////////
+   
+   public void CopyDir(string source, string dest)
+   {
+      DirectoryInfo dir = new DirectoryInfo(source);
+      
+      if (!dir.Exists)
+      {
+         string err_msg = $"Source directory {source} does not exit";
+         throw new DirectoryNotFoundException(err_msg);
+      }
+      
+      Directory.CreateDirectory(dest);
+      FileInfo[] files = dir.GetFiles();
+      
+      foreach (var file in files)
+      {
+         // Path to the new location.
+         string tempPath = Path.Combine(dest, file.Name);
+         
+         // Copy, overwrite if found.
+         file.CopyTo(tempPath, true);
+      }
+      
+      // Copy all sub directories.
+      DirectoryInfo[] dirs = dir.GetDirectories();
+      foreach (var subDir in dirs)
+      {
+         // Path to dest subdir
+         string tempPath = Path.Combine(dest, subDir.Name);
+         
+         // Recursive call to copy all files in the subDir
+         CopyDir(subDir.FullName, tempPath);
+      }
+   }
+
+   public void DownloadTests()
+   {
+      Directory.CreateDirectory(this.m_testRoot);
+
+      // Find the last successful test build based on OS.
+      string url = "http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/release_windows_nt/lastSuccessfulBuild/artifact/bin/tests/tests.zip";
```

Comment:
The cijobs tool does some of this today, might be better to not duplicate functionality.